### PR TITLE
Revert "Implement parsing XML document into xmerl format (#35)"

### PR DESCRIPTION
**Pull request by qcam**
This reverts commit 6ca0993f5a72fa584628d4c384a9271cbb8b6f05.

### CODE REVIEW

**Comment by timrogers** [comment]
@qcam Could you give a bit more detail on why you ended up reverting this? Thanks! 🙏

**Comment by qcam** [comment]
@timrogers There were some reasons as I mentioned in the description: moderate performance and memory leakage risks. More importantly, they can't be improved. `xmerl_xpath` requires  element data to be in char list while Saxy works with binary, the conversion between them will be eating up memory and degrading performance. `xmerl_xpath` also needs atom element names, which means atoms will be generated dynamically during parsing time.

Basically I didn't feel comfortable maintaining the feature in this shape unless someone would come up with better alternatives. I seek your understanding 🙏!

I'm totally fine if anyone takes this code and publishes it as a package though.

**Comment by qcam** [comment]
I ended up publishing this as a package myself. Please check out: https://github.com/qcam/saxmerl
